src/reconcile/rec_visibility.c

```diff
@@ -159,7 +159,7 @@ __rec_need_save_upd(WT_SESSION_IMPL *session, WT_UPDATE *selected_upd, uint64_t
   wt_timestamp_t max_ts, bool list_uncommitted, uint64_t flags)
 {
     /* Always save updates for in-memory database. */
-    if (LF_ISSET(WT_REC_IN_MEMORY))
+    if (LF_ISSET(WT_REC_IN_MEMORY) && !__wt_txn_visible_all(session, max_txn, max_ts))
```

Comment:
Can we do

if (LF_ISSET(WT_REC_IN_MEMORY) && __wt_txn_visible_all(session, max_txn, max_ts))
      return false;

if (LF_ISSET(WT_REC_IN_MEMORY))
      return true;

No need to call __wt_txn_visible_all twice if LF_ISSET(WT_REC_IN_MEMORY) && !__wt_txn_visible_all(session, max_txn, max_ts) is false.

Reply:
Also update the comment

Reply:
if (LF_ISSET(WT_REC_IN_MEMORY)
return !__wt_txn_visible_all(session, max_txn, max_ts);

Would be better.